src/services/accounts/AccountsStakingPayoutsService.ts

```diff
@@ -55,105 +88,202 @@ export class AccountsStakingPayoutsService extends AbstractService {
 					'than or equal to current_era - history_depth.'
 			);
 		}
+
 		const at = {
 			height: number.unwrap().toString(10),
 			hash,
 		};
 
-		const erasPayouts: (IEraPayouts | { message: string })[] = [];
-
 		// User friendly - we don't error if the user specified era & depth combo <= 0, instead just start at 0
 		const startEra = era - (depth - 1) < 0 ? 0 : era - (depth - 1);
 
-		for (let e = startEra; e <= era; e += 1) {
-			erasPayouts.push(
-				await this.deriveEraPayouts(
-					api,
-					hash,
+		// Fetch general data about the era
+		const allErasGeneral = await this.fetchAllErasGeneral(
+			api,
+			hash,
+			startEra,
+			era
+		);
+
+		// With the general data, we can now fetch the commission of each validator `address` nominates
+		const allErasCommissions = await this.fetchAllErasCommissions(
+			api,
+			hash,
+			address,
+			startEra,
+			allErasGeneral.map((el) => el[0])
```

Comment:
A bit lost here: what is in `el[0]` here? 
Did I get this right? 
- `allErasGeneral` is a `Promise.all` of a collection of `Promise.all`s, each making 3 RPC queries; 
- we `await` it so all promises are resolved and at this point `el[0]` is the result of calling `api.derive.staking.eraExposure(eraIndex)`
-  the other two RPCs are used later and returned as part of `allEraData`

It's a bit unfortunate that we have to loop over this collection twice, once here and then again in the `map()` call on line 118. Not sure what to suggest though.

Reply:
> It's a bit unfortunate that we have to loop over this collection twice, once here and then again in the map() call on line 118. Not sure what to suggest though.

1) I am not too concerned. ATM this array will never be over 84, so while technically linear to depth its complexity has a hard bound